### ISSUE TITLE
[ACTION NEEDED] Fix flaky integration tests at distribution level

### ISSUE THREAD

**Issue by gaiksaya** [opensearch-project/neural-search#667]
**What is the bug?**
It was observed in 2.13.0 and previous other releases that this component manually signed off on the release for failing integration tests. See https://github.com/opensearch-project/opensearch-build/issues/4433#issuecomment-2026436873
The flakiness of the test runs take a lot of time from the release team to collect go/no-go decision and significantly lower the confidence in the release bundles.

**How can one reproduce the bug?**
1. Run integration testing for altering and see the failures. 
2. Issues can be reproduced using the steps declared in AUTOCUT issues for failed integration testing

**What is the expected behavior?**
Tests should be consistently passing.

**Do you have any additional context?**
Please note that this is a hard blocker for 2.14.0 release as per the discussion [here](https://github.com/opensearch-project/opensearch-build/issues/4588)

**Comment by bbarani** [comment]
@martin-gaievski @vibrantvarun  Can you please provide your inputs?

**Comment by martin-gaievski** [comment]
As per our deep dive inconsistency in integ tests is caused by the implementation of memory circuit breaker on ml-commons side. In scope of our tests we're deploying/undeploying local models using ml-commons API and after multiple such calls memory CB became open. I've opened an issue in ml-commons for this matter https://github.com/opensearch-project/ml-commons/issues/2308. 
As a short-term mitigation for 2.14 we have optimized our tests (issues https://github.com/opensearch-project/neural-search/pull/683 and https://github.com/opensearch-project/neural-search/issues/689) to minimize number of times local model got redeployed. As of now integ tests are constantly pass in our local copy of https://github.com/opensearch-project/opensearch-build/
